src/main/solution_space/path_finder_test_system.py

```diff
@@ -0,0 +1,204 @@
+    # If some class doesn't have a method, there is self._no_method_sign (for example, '---') in a corresponding cell.
+    # If some class does have a method, but without docs, there is an empty cell
+    # +---------------+---------------+---------------+
+    # |  class name   |    method_1   |    method_2   |
+    # +---------------+---------------+---------------+
+    # |    class_1    |               |     *doc*     |
+    # +---------------+---------------+---------------+
+    # |    class_2    |     *doc*     |     ---       |
+    # +---------------+---------------+---------------+
+    # methods_to_keep should contain object methods, which should be included in the table.
+    # By default, all object methods are removed.
+    @staticmethod
+    def get_methods_doc_table(classes: List[Type[Class]], title: str,
+                              methods_to_keep: Optional[List[str]]) -> PrettyTable:
+        methods_dict, methods_names = TestSystem.__get_methods_dict_and_names(classes, methods_to_keep)
+        table = PrettyTable(field_names=['class name'] + methods_names, title=title)
+        for c in classes:
+            class_row = [c.__name__]
+            for name in methods_names:
+                method = methods_dict[c].get(name)
+                doc = TestSystem._no_method_sign if method is None else TestSystem.__format_doc_str(method.__doc__)
+                class_row.append(doc)
+            table.add_row(class_row)
+        return TestSystem.__set_table_style(table)
+
+    # Get a table with all path_finder results for a given test_inputs:
+    # +---------+-----+-------+---------------+---------------+
+    # | ... input headers...  | path_finder_1 | path_finder_2 |
+    # +---------+-----+-------+---------------+---------------+
+    # |     test_input_1      |   *result*    |   *result*    |
+    # +---------+-----+-------+---------------+---------------+
+    # |     test_input_2      |   *result*    |   *result*    |
+    # +---------+-----+-------+---------------+---------------+
+    def get_result_table(self, title: str) -> PrettyTable:
+        path_finders = self.__get_all_possible_path_finders()
+        # Set table headers: first go test_input headers, then path_finder versions
+        table = PrettyTable(field_names=[e.value for e in TEST_INPUT] +
+                                        [self.__get_path_finder_version(pf) for pf in path_finders], title=title)
+
+        for test_input in self._test_inputs:
+            user_vertex = self.__create_user_vertex(test_input)
+            row = [test_input[key] for key in TEST_INPUT]
+            for path_finder in path_finders:
+                row.append(self.__run_path_finder(path_finder, user_vertex))
+            table.add_row(row)
+
+        return TestSystem.__set_table_style(table)
+
+    def __get_all_possible_path_finders(self) -> List[IPathFinder]:
+        path_finders = []
+        for path_finder_subclass in self._path_finder_subclasses:
```

Comment:
Please, use **itertools.product**: https://stackoverflow.com/questions/16384109/iterate-over-all-combinations-of-values-in-multiple-lists-in-python